flutter/lib/src/screenshot/screenshot_attachment_processor.dart

```diff
@@ -0,0 +1,77 @@
+import 'dart:async';
+import 'dart:typed_data';
+import 'dart:ui' as ui show ImageByteFormat;
+
+import 'package:flutter/rendering.dart';
+import 'package:sentry/sentry.dart';
+import 'package:sentry/sentry_private.dart';
+import '../integrations/native_app_start_integration.dart';
+import '../renderer/renderer.dart';
+import '../sentry_flutter_options.dart';
+import 'sentry_screenshot_widget.dart';
+
+// ignore: invalid_use_of_internal_member
+class ScreenshotAttachmentProcessor implements SentryClientAttachmentProcessor {
+  final SchedulerBindingProvider _schedulerBindingProvider;
+  final SentryFlutterOptions _options;
+
+  ScreenshotAttachmentProcessor(this._schedulerBindingProvider, this._options);
+
+  /// This is true when the SentryWidget is in the view hierarchy
+  bool get _attachScreenshot =>
+      sentryScreenshotWidgetGlobalKey.currentContext != null;
+
+  @override
+  Future<List<SentryAttachment>> processAttachments(
+      List<SentryAttachment> attachments, SentryEvent event) async {
+    if (event.exceptions == null &&
+        event.throwable == null &&
+        _attachScreenshot) {
+      return attachments;
+    }
+    final renderer = _options.rendererWrapper.getRenderer();
+    if (renderer != FlutterRenderer.skia &&
+        renderer != FlutterRenderer.canvasKit) {
+      return attachments;
+    }
+
+    final schedulerBinding = _schedulerBindingProvider();
+    if (schedulerBinding != null) {
+      final completer = Completer<Uint8List?>();
+      schedulerBinding.addPostFrameCallback((timeStamp) async {
+        final screenshot = await _createScreenshot();
+        completer.complete(screenshot);
+      });
+      final bytes = await completer.future;
+      if (bytes == null) {
+        return attachments;
+      }
+      return attachments + [SentryAttachment.fromScreenshotData(bytes)];
+    } else {
+      return attachments;
+    }
+  }
+
+  Future<Uint8List?> _createScreenshot() async {
+    try {
+      final renderObject =
+          sentryScreenshotWidgetGlobalKey.currentContext?.findRenderObject();
+
+      if (renderObject is RenderRepaintBoundary) {
+        final image = await renderObject.toImage(pixelRatio: 1);
+        // At the time of writing there's no other image format available which
+        // Sentry understands.
+        final bytes = await image.toByteData(format: ui.ImageByteFormat.png);
```

Comment:
What's the average size for `png` here?
Maybe it'd be need to do the trade off here of quality?
See https://github.com/getsentry/sentry-java/blob/a04f78887caef1e188c16867cf2f068dd98e3017/sentry-android-core/src/main/java/io/sentry/android/core/ScreenshotEventProcessor.java#L83-L85

Reply:
From the documentation:

`
The returned ui.Image has uncompressed raw RGBA bytes in the dimensions of the render object, multiplied by the pixelRatio. Logical pixels are also referred to as device-independent or resolution-independent pixels.
`

In the test for example, the resolution of the image is `800x600`, for a total of `800*600 / 1024 = 468,75 kB`. What are typical image sizes on Android with the compression enabled?

Reply:
Ah, seems like the Images on android are typically around ~150 kB accordint to a comment we made. https://github.com/getsentry/sentry-java/blob/a04f78887caef1e188c16867cf2f068dd98e3017/sentry-android-core/src/main/java/io/sentry/android/core/ScreenshotEventProcessor.java#L97

Reply:
Searched a bit, it seems that many third party libraries rely on native implementations for image compression, using the native bridge, for example [`flutter_image_compress`](https://pub.dev/packages/flutter_image_compress).

There are also pure dart implementations, but they have more functionality then we need. We could extract what we need, for example a [JPG encoder](https://github.com/brendan-duncan/image/blob/23c147de111b9320aa45775b70b8c7a37e05732a/lib/src/formats/jpeg_encoder.dart) or a [PNG encoder](https://github.com/brendan-duncan/image/blob/23c147de111b9320aa45775b70b8c7a37e05732a/lib/src/formats/png_encoder.dart), provided that the license permits this.

Reply:
You can reduce the image size by changing the pixel ratio.

Reply:
Ah yes, i assumed 1 was the lowest value as it is the default. This results in an even smaller image. However, my math above is not correct in our case, as we use the PNG format and not `rawRgba`. So we get about 60 kB per image on an iOS Simulator. Think this is small enough for our purposes? Smaller images are harder to read.

<img width="709" alt="Bildschirmfoto 2022-11-08 um 15 19 53" src="https://user-images.githubusercontent.com/3984453/200589264-3ef3b953-b799-4c96-bf83-d27f0a1d031c.png">

Reply:
pixel ratio 1 and ~60kb is great, no need to vendor a lib.